static/js/components/inputs/SelectField.js

```diff
@@ -110,12 +78,15 @@ export default class SelectField extends React.Component {
     if (index === -1) {
       // enter was pressed and optionOrString is a string
       // select first item in dropdown if any are present
-      let filteredOptionValues = options.
-        map(option => option.label).
-        filter(this.autocompleteFilter.bind(this, optionOrString));
-      if (filteredOptionValues.length > 0) {
-        let option = options.find(option => option.label === filteredOptionValues[0]);
-        toStore = option.value;
+      let autocompleteProps = this.getAutocompleteProps();
+      let filterFunction = _.has(autocompleteProps, 'filter') ?
+        autocompleteProps.filter :
+        AutoCompleteSettings.defaultFilter;
+      let firstMatchingOption = _.find(options, (option) => {
```

Comment:
Can you replace `_.find(options` with `options.find(`, and the `{ return filterFunction(...); }` with `filterFunction(...)`?